File: UWP/Shiba/ViewMappers/ComponentMapper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Shiba.Controls;

namespace Shiba.ViewMappers
{
    internal class ComponentMapper : ViewMapper<ShibaHost>
    {
        protected override ShibaHost Map(View view, IShibaContext context)
        {
            var target = CreateNativeView(context);

            foreach (var property in view.Properties)
            {
                var cache = _propertyCache.Value.LastOrDefault(it => it.Name == property.Name);
                if (cache != null)
                {
                    SetValue(context, property.Value, cache, target);
                }
            }

            return target;
        }

        protected override IEnumerable<IValueMap> ExtendPropertyMap()
        {
            yield return new ManuallyValueMap("componentName", typeof(string), (element, o) =>
            {
                if (element is ShibaHost shibaHost && o is string value)
                {
                    shibaHost.Component = value;
                }
            });
            yield return new PropertyMap("dataContext", ShibaHost.DataContextProperty, typeof(object), typeof(object));
        }
    }
}

File: Android/shiba/src/main/java/org/shibajs/shiba/mapper/ComponentMapper.kt
package org.shibajs.shiba.mapper

import org.shibajs.shiba.IShibaContext
import org.shibajs.shiba.ShibaHost
import org.shibajs.shiba.type.ShibaExtension

class ComponentMapper : ViewMapper<ShibaHost>() {
    override fun createNativeView(context: IShibaContext): ShibaHost {
        return ShibaHost(context.getContext())
    }

    override fun convertSetValueType(propertyMap: PropertyMap, value: Any?, context: IShibaContext): Any? {
        return if (propertyMap.name != "dataContext") {
            super.convertSetValueType(propertyMap, value, context)
        } else {
            if (value is ShibaExtension) {
                super.convertSetValueType(propertyMap, value, context)
            } else {
                value
            }
        }
    }
    override fun propertyMaps(): ArrayList<PropertyMap> {
        return super.propertyMaps().apply {
            add(PropertyMap("componentName", { view, value ->
                if (view is ShibaHost && value is String) {
                    view.component = value
                }
            }))
            add(PropertyMap("dataContext", { view, value ->
                if (view is ShibaHost) {
                    view.dataContext = value
                }
            }))
        }
    }
}